src/web/mjs/connectors/Cycomi.mjs

```diff
@@ -8,54 +8,133 @@ export default class CyComi extends Connector {
+        const title = (await this.fetchDOM(request, 'h2.css-m40re5'))[0].textContent.trim();
         return new Manga(this, id, title);
     }
 
     async _getMangas() {
+        const uri = new URL('/api/title/serialization/list', this.url);
+        const request = new Request(uri, this.requestOptions);
+        const response = await this.fetchJSON(request, 'div.card-content > a.card');
         const mangaList = [];
-        for(let page = 0; page < 8; page++) {
-            const mangas = await this._getMangasFromPage(page);
+        for(let page = 0; page < 8; page += 1) {
+            const mangas = response.data.serialization[""+page].titles.map(title => {
+                return {
+                    id: title.titleId,
+                    title: title.titleName
+                };
+            });
             mangaList.push(...mangas);
         }
         return mangaList;
     }
 
-    async _getMangasFromPage(page) {
-        const uri = new URL('/fw/cycomibrowser/title/serialization/' + page, this.url);
-        const request = new Request(uri, this.requestOptions);
-        const data = await this.fetchDOM(request, 'div.card-content > a.card');
-        return data.map(element => {
-            return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.querySelector('.card-texts-title').textContent.trim()
-            };
-        });
+    async _getChapters(manga) {
+        let cursor = undefined;
+        const chapterList = [];
+        do {
+            const [chapters, nextCursor] = await this._getChaptersFromCursor(manga, cursor);
+            chapterList.push(...chapters);
+            cursor = nextCursor;
+        } while (cursor);
+        return chapterList;
     }
 
-    async _getChapters(manga) {
-        const uri = new URL(manga.id, this.url);
+    async _getChaptersFromCursor(manga, cursor) {
+        const params = new URLSearchParams({
+            limit: 20,
+            titleId: manga.id,
+            sort: 2,
+        });
+        if (cursor) {
+            params.append("cursor", cursor);
+        }
+        const uri = new URL("/api/chapter/paginatedList?" + params.toString(), this.url);
         const request = new Request(uri, this.requestOptions);
-        const data = await this.fetchDOM(request, '.title-chapters > a');
-        return data.map(element => {
+        const response = await this.fetchJSON(request);
+        const chapters = response.data.map(element => {
             return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.querySelector('.chapter-title').textContent.trim(),
+                id: {"chapterId": element.id, "titleId": manga.id},
+                title: element.name + " " + element.subName,
                 language: '',
             };
-        }).reverse();
+        });
+        return [chapters, response.nextCursor];
+    }
+
+    s(e, t) {
```

Comment:
I am not an expert but this looks like RC4 to me (256 box and xor operation). Maybe it could be replaced using the avaiblable CryptoJS code? Well at least rename it to something better than "s" :)